src/main/java/com/autotune/analyzer/utils/ExperimentResultValidation.java

```diff
@@ -55,18 +63,56 @@ public void validate(List<ExperimentResultData> experimentResultDataList) {
                             resultData.setValidationResultData(new ValidationResultData(false, errorMsg));
                             break;
                         }
-                        // Validate Performance Profile data
-                        RemoteMonitoringOpenShiftImpl remoteMonitoringOpenShiftImpl = new RemoteMonitoringOpenShiftImpl();
-                        errorMsg = remoteMonitoringOpenShiftImpl.validate(kruizeObject,resultData);
-                        if (errorMsg.isEmpty() || errorMsg.isBlank()) {
-                        // call recommend() method here
-                            remoteMonitoringOpenShiftImpl.recommend();
-                            proceed = true;
-                        } else {
+                        /*
+                         Fetch the performance profile from the Map corresponding to the name in the kruize object,
+                         and then validate the Performance Profile data
+                        */
+                        try {
+                            PerformanceProfile performanceProfile = performanceProfilesMap.get(kruizeObject.getPerformanceProfile());
+                            Class<?> validationClass = null;
+                            Class<?>[] parameterTypes = null;
+                            Method method = null;
+                            Object object = null;
+                            // check if performance profile is absent and then attach a default one
+                            if (null == performanceProfile) {
+                                SloInfo sloInfo = kruizeObject.getSloInfo();
+                                performanceProfile = new PerformanceProfile(AnalyzerConstants.PerformanceProfileConstants.DEFAULT_PROFILE,AnalyzerConstants.DEFAULT_PROFILE_VERSION, AnalyzerConstants.DEFAULT_K8S_TYPE, sloInfo);
+                                kruizeObject.setPerformanceProfile(performanceProfile.getName());
+                                errorMsg = new DefaultOpenshiftImpl().validate(performanceProfile, resultData);
```

Comment:
Why is this being called `DefaultOpenshiftImpl`? This has nothing to do with OpenShift. It is a generic profile for all experiments that don't define a profile

Reply:
Updated